CI: Conformance AWS-CNI: creating netlink handle: failed to set into network namespace while creating netlink socket: invalid argument

## CI failure

```
📋 Test Report [cilium-test-1]
❌ 1/2 tests failed (1/29 actions), 6 tests skipped, 0 scenarios skipped:
Test [check-log-errors]:
  🟥 check-log-errors/no-errors-in-logs:vendor/github.com/cilium/hive:cilium-cilium-16125135816-1.us-east-1.eksctl.io/kube-system/cilium-mzg8g (cilium-agent): Found 3 logs in cilium-cilium-16125135816-1.us-east-1.eksctl.io/kube-system/cilium-mzg8g (cilium-agent) from before pod restart matching list of errors that must be investigated:
time=2025-07-07T18:57:01.280671647Z level=error source=/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/cell/lifecycle.go:129 msg="Start hook failed" function="neighbor.newNetlinkFuncsGetter.func1 (.../datapath/neighbor/neighbor_reconciler.go:44) (agent.datapath.neighbor)" error="creating netlink handle: failed to set into network namespace 49 while creating netlink socket: invalid argument" (1 occurrences)
time=2025-07-07T18:57:01.280727844Z level=error source=/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:363 msg="Failed to start hive" error="creating netlink handle: failed to set into network namespace 49 while creating netlink socket: invalid argument" duration=3.896069564s (1 occurrences)
panic: Start or stop failed to finish on time, aborting forcefully. (1 occurrences)
    ⛑️ The following owners are responsible for reliability of the testsuite: 
        - @cilium/vendor (no-errors-in-logs:vendor/github.com/cilium/hive)
        - @cilium/aws (.github/workflows/conformance-aws-cni.yaml)
        - @cilium/ci-structure (.github/workflows/conformance-aws-cni.yaml)
[cilium-test-1] 1 tests failed
```

Link: https://github.com/cilium/cilium/actions/runs/16125135816/job/45515316703
Sysdump: No sysdump produced

--- a/pkg/datapath/neighbor/neighbor_reconciler.go
+++ b/pkg/datapath/neighbor/neighbor_reconciler.go
@@ -15,11 +15,9 @@ import (
 	"github.com/cilium/statedb"
 	"github.com/cilium/statedb/reconciler"
 	"github.com/vishvananda/netlink"
-	vns "github.com/vishvananda/netns"
 	"golang.org/x/sys/unix"
 
 	"github.com/cilium/cilium/pkg/datapath/tables"
-	"github.com/cilium/cilium/pkg/netns"
 	"github.com/cilium/cilium/pkg/rate"
 	"github.com/cilium/cilium/pkg/time"
 )
@@ -42,15 +40,10 @@ func newNetlinkFuncsGetter(lifecycle cell.Lifecycle) *netlinkFuncsGetter {
 	lifecycle.Append(
 		cell.Hook{
 			OnStart: func(_ cell.HookContext) error {
-				// Get a netlink handle in the [netns.Current] namespace.
-				// Otherwise we default to the host namespace. Which is not what we want
+				// Get a netlink handle in the current namespace.
+				// Otherwise we default to the namespace at startup. Which is not what we want
 				// during testing where we might currently be in a sub-namespace.
-				cur, err := netns.Current()
-				if err != nil {
-					return fmt.Errorf("getting current netns: %w", err)
-				}
-
-				handle, err := netlink.NewHandleAt(vns.NsHandle(cur.FD()))
+				handle, err := netlink.NewHandle()
 				if err != nil {
 					return fmt.Errorf("creating netlink handle: %w", err)
 				}

